### PR TITLE
Akhrorbek

### PR DESCRIPTION
**Pull request by AkhrorbekAbdurakhimov**

### CODE REVIEW

**Comment by AkhrorbekAbdurakhimov** [comment]
Dear Dzernakou Uladzislau.
I send request to collaborate my backend side. Please accept my request and check project with backed part
I think I did the homework 11, but didn't tick in table, or have I misunderstanding in homework

**Comment by vokanred** [comment]
> Dear Dzernakou Uladzislau. I send request to collaborate my backend side. Please accept my request and check project with backed part I think I did the homework 11, but didn't tick in table, or have I misunderstanding in homework

I don't understand what homework 11 is. What do you mean?

**Comment by vokanred** [comment]
Please add Logout button

**Comment by vokanred** [comment]
If you decided to use postgresql why didn't you create migrations to create DB?
Don't understand how I should create it.

File: client/src/app/categories-page/categories-page.component.html [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<div class="container">
```
Comment (vokanred) [inline client/src/app/categories-page/categories-page.component.html:1]: This page should be split into components

File: client/src/app/categories-page/categories-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        .addCategory(categoryType, title)
+        .pipe(untilDestroyed(this))
+        .subscribe({
+          next: data => {
+            this.toasterMessageStatus = true;
+            this.toasterMessage = data.message;
+            this.isOpenAddCategoryModal = false;
+            setTimeout(() => {
+              this.toasterMessageStatus = false;
+              window.location.reload()
```
Comment (vokanred) [inline client/src/app/categories-page/categories-page.component.ts]: You shouldn't reload the page.

File: client/src/app/categories-page/categories-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      .deleteCategory(this.deletedCategoryId)
+      .pipe(untilDestroyed(this))
+      .subscribe({
+        next: data => {
+          this.isWarning = false;
+          this.toasterMessage = data.message;
+          this.toasterMessageStatus = true;
+          setTimeout(() => {
+            this.toasterMessageStatus = false;
+            window.location.reload();
```
Comment (vokanred) [inline client/src/app/categories-page/categories-page.component.ts]: Don't reload

File: client/src/app/main-page/main-aside/main-aside.component.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <div class="container">
```
Comment (vokanred) [inline client/src/app/main-page/main-aside/main-aside.component.html:1]: Should be split to components

File: client/src/app/main-page/main-aside/main-aside.component.ts [hunk 5]
```diff
@@ -54,14 +107,77 @@ export class MainAsideComponent implements OnInit {
...
+            if (data.type === 'warning') {
+              this.isWarning = true;
+              this.toasterMessageText = data.message;
+              this.toasterMessageStatus = true;
+              setTimeout(() => {
+                this.toasterMessageStatus = false;
+              }, 2500)
+            } else {
+              this.isWarning = false;
+              window.location.reload();
```
Comment (vokanred) [inline client/src/app/main-page/main-aside/main-aside.component.ts:151]: don't reload

**Review by vokanred — CHANGES_REQUESTED** [review]
Do not use reload()
Do not create pages with 100+ lines of code

Looks like you try to integrate existing HTML layout with Angular...

**Comment by vokanred** [comment]
Tests are done.

**Comment by AkhrorbekAbdurakhimov** [comment]
> 



> > Dear Dzernakou Uladzislau. I send request to collaborate my backend side. Please accept my request and check project with backed part I think I did the homework 11, but didn't tick in table, or have I misunderstanding in homework
> 
> I don't understand what homework 11 is. What do you mean?

Sorry, homework 8, 11 april

**Comment by AkhrorbekAbdurakhimov** [comment]
> If you decided to use postgresql why didn't you create migrations to create DB? Don't understand how I should create it.

I have already created database in [elepehent](https://www.elephantsql.com/) . You shouldn't create, I configure my database service as a server in this server [elepehent](https://www.elephantsql.com/). You must  just clone my backend side and just run it. If you want I upload my project to heroku

**Comment by AkhrorbekAbdurakhimov** [comment]
> Please add Logout button

Ok I will add

**Comment by vokanred** [comment]
> > If you decided to use postgresql why didn't you create migrations to create DB? Don't understand how I should create it.
> 
> I have already created database in [elepehent](https://www.elephantsql.com/) . You shouldn't create, I configure my database service as a server in this server [elepehent](https://www.elephantsql.com/). You must just clone my backend side and just run it. If you want I upload my project to heroku

Pls provide some test user credentials

File: client/src/app/categories-page/categories-page.component.html (hunk 1 above)
Comment (AkhrorbekAbdurakhimov) [reply client/src/app/categories-page/categories-page.component.html:1]: Ok

File: client/src/app/categories-page/categories-page.component.ts (hunk 2 above)
Comment (AkhrorbekAbdurakhimov) [reply client/src/app/categories-page/categories-page.component.ts]: When I use route.navigateByUrl() it was not working So I used window.location.reload() but I will try to solve this bug

File: client/src/app/categories-page/categories-page.component.ts (hunk 3 above)
Comment (AkhrorbekAbdurakhimov) [reply client/src/app/categories-page/categories-page.component.ts]: I will use route.navigateByUrl('')

File: client/src/app/main-page/main-aside/main-aside.component.html (hunk 4 above)
Comment (AkhrorbekAbdurakhimov) [reply client/src/app/main-page/main-aside/main-aside.component.html:1]: Ok

File: client/src/app/main-page/main-aside/main-aside.component.ts (hunk 5 above)
Comment (AkhrorbekAbdurakhimov) [reply client/src/app/main-page/main-aside/main-aside.component.ts:151]: OK

**Comment by AkhrorbekAbdurakhimov** [comment]
> > > If you decided to use postgresql why didn't you create migrations to create DB? Don't understand how I should create it.
> > 
> > 
> > I have already created database in [elepehent](https://www.elephantsql.com/) . You shouldn't create, I configure my database service as a server in this server [elepehent](https://www.elephantsql.com/). You must just clone my backend side and just run it. If you want I upload my project to heroku
> 
> Pls provide some test user credentials

email-> akhrorbek20011707@gmail.com
password -> password12345#

**Comment by AkhrorbekAbdurakhimov** [comment]
Dear  Dzernakou Uladzislau.
My sphere is backend and I am newby in angular, I don't know standards of fronted but I will try to learn.
I try to split pages to components but it requires more time and presentation demo is near. firstly I try to complete main features. Then I will try split smaller components to my best.

File: client/src/app/categories-page/categories-page.component.ts (hunk 3 above)
Comment (vokanred) [reply client/src/app/categories-page/categories-page.component.ts]: I don't understand why you want to navigate

File: client/src/app/categories-page/categories-page.component.ts (hunk 2 above)
Comment (vokanred) [reply client/src/app/categories-page/categories-page.component.ts]: Pls solve

**Comment by vokanred** [comment]
I understand but one of the tasks is to split pages to components.

**Comment by vokanred** [comment]
At least fix the issue with reload. If something is not working feel free to ask questions

**Review by vokanred — APPROVED** [review]
Approved

**Pull request merged**